components/prism-pascaligo.js

```diff
@@ -0,0 +1,58 @@
+// Pascaligo is a layer 2 smart contract language for the tezos blockchain
+
+Prism.languages.pascaligo = {
+	'comment': [
+		/\(\*[\s\S]+?\*\)/,
+		/\/\/.*/
+	],
+	'string': [
+		{
+			pattern: /(?:'(?:''|[^'\r\n])*'|#[&$%]?[a-f\d]+)+|\^[a-z]/i,
+			greedy: true
+		},
+		{
+			pattern: /(?:"(?:""|[^"\r\n])*"|#[&$%]?[a-f\d]+)+|\^[a-z]/i,
+			greedy: true
+		},
+	],
+	'keyword': [
+		{
+			pattern: /(^|[^&])\b(?:if|while|for|return|nil|remove|from|else|then|skip|with|is|function|var|const|type|end|begin|block)\b/i,
+			lookbehind: true
+		},
+		{
+			pattern: /(^|[^&])\b(?:True|False)\b/i,
+			lookbehind: true
+		},
+    ],
+    'builtin': {
+        pattern: /(^|[^&])\b(?:int|unit|string|nat|map)\b/i,
+		lookbehind: true
+    },
+    'function': {
+        pattern: /(?!with\b)\b\w+ ?(?=\()/i,
```

Comment:
Shouldn't `with` be matched by `keyword` already? Can you give an example where you need the negative lookahead?

Reply:
Yeah this is mainly functions that look like:
```
function pop (const h : heap) : (heap * heap_element * nat) is
  begin
   const result : heap_element = get_top (h) ;
   var s : nat := size(h) ;
   const last : heap_element = get_force(s, h) ;
   remove s from map h ;
   h[1n] := last ;
   s := size(h) ;
   var i : nat := 0n ;
   var largest : nat := 1n ;
   var left : nat := 0n ;
   var right : nat := 0n ;
   var c : nat := 0n ;
   while (largest =/= i) block {
     c := c + 1n ;
     i := largest ;
     left := 2n * i ;
     right := left + 1n ;
     if (left <= s) then begin
       if (heap_element_lt(get_force(left , h) , get_force(i , h))) then begin
        largest := left ;
        const tmp : heap_element = get_force(i , h) ;
        h[i] := get_force(left , h) ;
        h[left] := tmp ;
       end else skip ;
     end else if (right <= s) then begin
       if (heap_element_lt(get_force(right , h) , get_force(i , h))) then begin
         largest := right ;
         const tmp : heap_element = get_force(i , h) ;
         h[i] := get_force(right , h) ;
         h[left] := tmp ;
       end else skip ;
     end else skip ;
  }
  end with (h , result , c)
```

specifically the end of the function  `with (h , result , c)`. This is a return statement and not to be identified as a function.

Reply:
Seems like I missed that `with` is a keyword already. In this case, you don't need the assertion at all.